From f8b820f7adfb9a58eca0c308c2ad97e022c7d301 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:05:09 -0400
Subject: [tools] perf/probe: Remove incorrect symbol check for --list

Message-id: <1410181752-25631-65-git-send-email-jolsa@redhat.com>
Patchwork-id: 91746
O-Subject: [PATCH RHEL7.1 BZ1134356 064/307] perf probe: Remove incorrect symbol check for --list
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit f49540b17c1c6fa5a0734cc1d8b57614fd2036be
Author: Masami Hiramatsu <masami.hiramatsu.pt@hitachi.com>
Date: Thu Feb 6 05:32:06 2014 +0000

description
===========
Remove unneeded symbol check for --list option.

This code actually checks whether the given symbol exists in the kernel.
But this is incorrect for online kernel/module and offline module too:

 - For online kernel/module, the kprobes itself already
  ensured the symbol exist in the kernel.
 - For offline module, this code can't access the offlined
  modules. Ignore it.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/probe-event.c b/tools/perf/util/probe-event.c
index 9aa7783..a4649e7 100644
--- a/tools/perf/util/probe-event.c
+++ b/tools/perf/util/probe-event.c
@@ -739,14 +739,6 @@ out:
 static int kprobe_convert_to_perf_probe(struct probe_trace_point *tp,
      struct perf_probe_point *pp)
 {
- struct symbol *sym;
-
- sym = __find_kernel_function_by_name(tp->symbol, NULL);
- if (!sym) {
-  pr_err("Failed to find symbol %s in kernel.\n", tp->symbol);
-  return -ENOENT;
- }
-
  return convert_to_perf_probe_point(tp, pp);
 }
 
-- 
1.7.1